main/src/cgeo/geocaching/activity/TokenAuthorizationActivity.java

```diff
@@ -0,0 +1,322 @@
+package cgeo.geocaching.activity;
+
+import butterknife.InjectView;
+
+import cgeo.geocaching.Intents;
+import cgeo.geocaching.R;
+import cgeo.geocaching.network.Network;
+import cgeo.geocaching.network.Parameters;
+import cgeo.geocaching.utils.BundleUtils;
+import cgeo.geocaching.utils.Log;
+import cgeo.geocaching.utils.MatcherWrapper;
+import cgeo.geocaching.utils.RxUtils;
+
+import ch.boye.httpclientandroidlib.HttpResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jdt.annotation.NonNull;
+
+import rx.functions.Action0;
+
+import android.app.Activity;
+import android.app.ProgressDialog;
+import android.content.ActivityNotFoundException;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Message;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import java.util.regex.Pattern;
+
+public abstract class TokenAuthorizationActivity extends AbstractActivity {
+
+    private static final Pattern PATTERN_IS_ERROR = Pattern.compile("error ([\\d]+)");
+    private static final Pattern PATTERN_TOKEN = Pattern.compile("([\\w]+)");
+
+    public static final int NOT_AUTHENTICATED = 0;
+    public static final int AUTHENTICATED = 1;
+    private static final int ERROR_EXT_MSG = 2;
+
+    @NonNull private String host = StringUtils.EMPTY;
+    @NonNull private String pathToken = StringUtils.EMPTY;
+    @NonNull private String urlRegister = StringUtils.EMPTY;
+    @NonNull private String fieldUsername = StringUtils.EMPTY;
+    @NonNull private String fieldPassword = StringUtils.EMPTY;
+
+    private boolean https = false;
```

Comment:
Better use something that can be identified as `boolean`, such as `useHTTPS`. Nevertheless, why single out http/https and pass it as an extra, rather than getting it from the URL? And why decompose the URL into components at all?